### PR TITLE
DDS PSM Core module files added

### PR DESCRIPTION
**Pull request by laura-eprosima**

### CODE REVIEW

**Comment by richiware** [comment]
Build status:
* Linux [Build Status](http://jenkins.eprosima.com:8080/job/FastRTPS%20Manual%20Linux/1593/)
* Mac [Build Status](http://jenkins.eprosima.com:8080/job/FastRTPS%20Manual%20Mac/1359/)
* Windows [Build Status](http://jenkins.eprosima.com:8080/job/FastRTPS%20Manual%20Windows/1589/)

**Comment by richiware** [comment]
Build status:
* Linux [Build Status](http://jenkins.eprosima.com:8080/job/FastRTPS%20Manual%20Linux/1596/)
* Mac [Build Status](http://jenkins.eprosima.com:8080/job/FastRTPS%20Manual%20Mac/1362/)
* Windows [Build Status](http://jenkins.eprosima.com:8080/job/FastRTPS%20Manual%20Windows/1592/)

**Comment by richiware** [comment]
Build status:
* Linux [Build Status](http://jenkins.eprosima.com:8080/job/FastRTPS%20Manual%20Linux/1600/)
* Mac [Build Status](http://jenkins.eprosima.com:8080/job/FastRTPS%20Manual%20Mac/1366/)
* Windows [Build Status](http://jenkins.eprosima.com:8080/job/FastRTPS%20Manual%20Windows/1596/)

File: include/dds/core/status/State.hpp [hunk 1]
```diff
@@ -0,0 +1,406 @@
...
+    {
+        *this |= mask;
+        return *this;
+    }
+
+    /**
+     * Shift (merge) given StatusMask bits into this StatusMask bitset.
+     *
+     * @return StatusMask this
+     */
```
Comment (IkerLuengo) [inline include/dds/core/status/State.hpp]: This method does not merge, but assign

File: include/dds/core/status/State.hpp [hunk 2]
```diff
@@ -0,0 +1,406 @@
...
+    inline StatusMask operator =(
+            const std::bitset<OMG_DDS_STATUS_COUNT>& mask) const
+    {
+        std::stringstream ss;
+        int number;
+        ss << std::hex << mask.to_ulong();
+        ss >> number;
+        return StatusMask(number);
```
Comment (IkerLuengo) [inline include/dds/core/status/State.hpp]: This code just returns a copy of the input mask, does not assign it to *this

File: include/dds/core/status/State.hpp [hunk 3]
```diff
@@ -0,0 +1,406 @@
...
+    {
+        bitset<16> r = *this & status;
+        return r == status;
+    }
+
+    bool is_compatible(
+            StatusMask base) const
+    {
+        return (*this == all() || *this == base);
+    }
```
Comment (IkerLuengo) [inline include/dds/core/status/State.hpp]: This method is not necessary,  `is_active` does the job better.

File: include/dds/core/status/State.hpp [hunk 4]
```diff
@@ -0,0 +1,406 @@
...
+     */
+    inline static StatusMask subscription_matched()
+    {
+        return StatusMask(0x00000001 << 14u);
+    }
+
+    bool is_active(
+            StatusMask status)
+    {
+        bitset<16> r = *this & status;
```
Comment (IkerLuengo) [inline include/dds/core/status/State.hpp]: Avoid magic numbers, use `OMG_DDS_STATUS_COUNT` instead of `16`

**Comment by richiware** [comment]
Build status:
* Linux [Build Status](http://jenkins.eprosima.com:8080/job/FastRTPS%20Manual%20Linux/1613/)
* Mac [Build Status](http://jenkins.eprosima.com:8080/job/FastRTPS%20Manual%20Mac/1379/)
* Windows [Build Status](http://jenkins.eprosima.com:8080/job/FastRTPS%20Manual%20Windows/1609/)

**Pull request merged**